Regression in Liquibase 4.25.0: Liquibase#calculateCheckSum() always fails

### Search first

- [X] I searched and no similar issues were found

### Description

The changes in https://github.com/liquibase/liquibase/pull/4463 cause `Liquibase#calculateCheckSum()` to always fail with the error message:

> ```
> Error encountered while parsing the command line. '--changeset-identifier' cannot be provided alongside other changeset arguments: '--changeset-id', '--changeset-path', '--changeset-author'.
> ```

Looks like this check is responsible for the error message:
https://github.com/liquibase/liquibase/blame/v4.25.1/liquibase-standard/src/main/java/liquibase/command/core/CalculateChecksumCommandStep.java#L140-L142

The command is always constructed with 

`CalculateChecksumCommandStep.CHANGESET_PATH_ARG`, `CalculateChecksumCommandStep.CHANGESET_ID_ARG`, `CalculateChecksumCommandStep.CHANGESET_AUTHOR_ARG`, *and* `CalculateChecksumCommandStep.CHANGESET_IDENTIFIER_ARG` which triggers the exception:

https://github.com/liquibase/liquibase/blob/v4.25.1/liquibase-standard/src/main/java/liquibase/Liquibase.java#L1271-L1278

### Steps To Reproduce

With Liquibase 4.25.0:
```java
@Test
void liquibasePullRequest4463() throws LiquibaseException {
  Liquibase liquibase = new Liquibase("migrations.xml", new ClassLoaderResourceAccessor(), new H2Database());
  liquibase.calculateCheckSum("", "1", "name");
}
```

Exception:
```
liquibase.exception.CommandExecutionException: liquibase.exception.LiquibaseException: java.lang.IllegalArgumentException: Error encountered while parsing the command line. '--changeset-identifier' cannot be provided alongside other changeset arguments: '--changeset-id', '--changeset-path', '--changeset-author'.

	at liquibase.command.CommandScope.execute(CommandScope.java:237)
	at liquibase.Liquibase.calculateCheckSum(Liquibase.java:1278)
	at com.github.joschi.liquibase.DbCalculateChecksumCommandTest.liquibasePullRequest4463(DbCalculateChecksumCommandTest.java:47)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
Caused by: liquibase.exception.LiquibaseException: java.lang.IllegalArgumentException: Error encountered while parsing the command line. '--changeset-identifier' cannot be provided alongside other changeset arguments: '--changeset-id', '--changeset-path', '--changeset-author'.
	at liquibase.command.core.CalculateChecksumCommandStep.validateIdentifierParameters(CalculateChecksumCommandStep.java:148)
	at liquibase.command.core.CalculateChecksumCommandStep.run(CalculateChecksumCommandStep.java:95)
	at liquibase.command.CommandScope.execute(CommandScope.java:214)
	... 5 more
Caused by: java.lang.IllegalArgumentException: Error encountered while parsing the command line. '--changeset-identifier' cannot be provided alongside other changeset arguments: '--changeset-id', '--changeset-path', '--changeset-author'.
	... 8 more
```

### Expected/Desired Behavior

The checksum of the changeset should be calculated without an exception.

### Liquibase Version

4.25.0

### Database Vendor & Version

All

### Liquibase Integration

Code (via liquibase.Liquibase)

### Liquibase Extensions

_No response_

### OS and/or Infrastructure Type/Provider

_No response_

### Additional Context

- https://github.com/dropwizard/dropwizard/pull/7883
- https://github.com/dropwizard/dropwizard/pull/7891
- https://github.com/dropwizard/dropwizard/pull/7896

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR (Thank you!)

--- a/liquibase-standard/src/main/java/liquibase/Liquibase.java
+++ b/liquibase-standard/src/main/java/liquibase/Liquibase.java
@@ -1273,7 +1273,6 @@ public class Liquibase implements AutoCloseable {
                 .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_PATH_ARG, changeSetPath)
                 .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_ID_ARG, changeSetId)
                 .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_AUTHOR_ARG, changeSetAuthor)
-                .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_IDENTIFIER_ARG, String.format("%s::%s::%s", changeSetPath, changeSetId, changeSetAuthor))
                 .addArgumentValue(CalculateChecksumCommandStep.CHANGELOG_FILE_ARG, this.changeLogFile)
                 .execute();
         return commandResults.getResult(CalculateChecksumCommandStep.CHECKSUM_RESULT);

--- a/liquibase-standard/src/main/java/liquibase/command/core/CalculateChecksumCommandStep.java
+++ b/liquibase-standard/src/main/java/liquibase/command/core/CalculateChecksumCommandStep.java
@@ -157,15 +157,15 @@ public class CalculateChecksumCommandStep extends AbstractCommandStep {
                     "If --changeset-identifier is not provided than --changeset-id, --changeset-author and --changeset-path must be specified. " +
                     "Missing argument: ";
 
-            if (commandScope.getArgumentValue(CHANGESET_ID_ARG) == null) {
+            if (StringUtil.isEmpty(commandScope.getArgumentValue(CHANGESET_ID_ARG))) {
                 errorMessage = errorMessage + " '--changeset-id',";
             }
 
-            if (commandScope.getArgumentValue(CHANGESET_AUTHOR_ARG) == null) {
+            if (StringUtil.isEmpty(commandScope.getArgumentValue(CHANGESET_AUTHOR_ARG))) {
                 errorMessage = errorMessage + " '--changeset-author',";
             }
 
-            if (commandScope.getArgumentValue(CHANGESET_PATH_ARG) == null) {
+            if (StringUtil.isEmpty(commandScope.getArgumentValue(CHANGESET_PATH_ARG))) {
                 errorMessage = errorMessage + " '--changeset-path',";
             }
 

--- a/liquibase-standard/src/test/groovy/liquibase/command/CalculateChecksumCommandStepTest.groovy
+++ b/liquibase-standard/src/test/groovy/liquibase/command/CalculateChecksumCommandStepTest.groovy
@@ -1,0 +1,120 @@
+package liquibase.command
+
+import liquibase.Scope
+import liquibase.command.core.CalculateChecksumCommandStep
+import liquibase.command.core.helpers.DbUrlConnectionCommandStep
+import liquibase.database.core.MockDatabase
+import liquibase.exception.CommandExecutionException
+import liquibase.resource.SearchPathResourceAccessor
+import liquibase.util.StringUtil
+import spock.lang.Specification
+import spock.lang.Unroll
+
+class CalculateChecksumCommandStepTest extends Specification {
+
+    def validateCheckSumIsCalculatedSuccessfully() {
+        when:
+        def generatedCheckSum = calculateCheckSum("tagged-changelog.xml", "1", "liquibase")
+
+        then:
+        StringUtil.isNotEmpty(generatedCheckSum.toString())
+    }
+
+    //Negative Tests
+    @Unroll
+    def "validate CheckSum is not calculated having a missing path argument"(String path) {
+        when:
+        def generatedCheckSum = calculateCheckSum(path, "1", "liquibase")
+
+        then:
+        StringUtil.isEmpty(generatedCheckSum)
+        def e = thrown(CommandExecutionException)
+        e.message.contains("Missing argument:  '--changeset-path'")
+
+        where:
+        path << ["", null]
+    }
+
+    @Unroll
+    def "validate CheckSum is not calculated having a missing author argument"(String author) {
+        when:
+        def generatedCheckSum = calculateCheckSum("tagged-changelog.xml", "1", author)
+
+        then:
+        StringUtil.isEmpty(generatedCheckSum)
+        def e = thrown(CommandExecutionException)
+        e.message.contains("Missing argument:  '--changeset-author'")
+
+        where:
+        author << ["", null]
+    }
+
+    @Unroll
+    def "validate CheckSum is not calculated having a missing Id argument"(String id) {
+        when:
+        def generatedCheckSum = calculateCheckSum("tagged-changelog.xml", id, "liquibase")
+
+        then:
+        StringUtil.isEmpty(generatedCheckSum)
+        def e = thrown(CommandExecutionException)
+        e.message.contains("Missing argument:  '--changeset-id'")
+
+        where:
+        id << ["", null]
+    }
+
+    def "validate error is returned when changesetIdentifier and the three separate arguments are provided at the same time"() {
+        when:
+        def resourceAccessor = new SearchPathResourceAccessor("target/test-classes/liquibase")
+        def scopeSettings = [
+                (Scope.Attr.resourceAccessor.name()): resourceAccessor
+        ]
+
+        def generatedCheckSum = ""
+
+        Scope.child(scopeSettings, {
+            CommandResults commandResults = new CommandScope("calculateChecksum")
+                    .addArgumentValue(CalculateChecksumCommandStep.CHANGELOG_FILE_ARG, "tagged-changelog.xml")
+                    .addArgumentValue(DbUrlConnectionCommandStep.DATABASE_ARG, new MockDatabase())
+                    .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_PATH_ARG, "tagged-changelog.xml")
+                    .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_ID_ARG, "1")
+                    .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_AUTHOR_ARG, "liquibase")
+                    .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_IDENTIFIER_ARG, "tagged-changelog.xml::1::liquibase")
+                    .execute()
+            generatedCheckSum = commandResults.getResult("checksumResult")
+        } as Scope.ScopedRunner)
+
+        then:
+        StringUtil.isEmpty(generatedCheckSum)
+        def e = thrown(CommandExecutionException)
+        e.message.contains("'--changeset-identifier' cannot be provided alongside other changeset arguments: '--changeset-id', '--changeset-path', '--changeset-author'")
+    }
+
+    private String calculateCheckSum(String path, String id, String author) {
+        try {
+            def resourceAccessor = new SearchPathResourceAccessor("target/test-classes/liquibase")
+            def scopeSettings = [
+                    (Scope.Attr.resourceAccessor.name()): resourceAccessor
+            ]
+
+            def generatedCheckSum = ""
+
+            Scope.child(scopeSettings, {
+                CommandResults commandResults = new CommandScope("calculateChecksum")
+                        .addArgumentValue(CalculateChecksumCommandStep.CHANGELOG_FILE_ARG, "tagged-changelog.xml")
+                        .addArgumentValue(DbUrlConnectionCommandStep.DATABASE_ARG, new MockDatabase())
+                        .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_PATH_ARG, path)
+                        .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_ID_ARG, id)
+                        .addArgumentValue(CalculateChecksumCommandStep.CHANGESET_AUTHOR_ARG, author)
+                        .execute()
+                generatedCheckSum = commandResults.getResult("checksumResult")
+            } as Scope.ScopedRunner)
+
+            return generatedCheckSum
+        }
+        catch (CommandExecutionException e) {
+            throw e
+        }
+
+    }
+}

